### PR TITLE
Ability to mark/flag email as spam, junk, or phishing

### LINKED ISSUE #2430: [Feature suggestion] Ability to mark/flag email as spam, junk, or phishing
Description: We have received a feature suggestion for the Android app: the ability to flag/move emails as spam, junk, or phishing. They find the said feature to be helpful.

Reference: https://mail.google.com/mail/u/human@flowcrypt.com/#inbox/FMfcgzGtwqFGqfvtRLkHrmrbQNwtwVbb

### PR DESCRIPTION
**Pull request by DenBond7**
This PR add ability to mark/flag an email as spam, junk, or phishing

close #2430

----------------------------------

**Tests** _(delete all except exactly one)_:
- Does not need tests (refactor only, docs or internal changes)

--------------------------------

### To be filled by reviewers

I have reviewed that this PR... _(tick whichever items you personally focused on during this review)_:
- [ ] addresses the issue it closes (if any)
- [ ] code is readable and understandable
- [ ] is accompanied with tests, or tests are not needed
- [ ] is free of vulnerabilities

### CODE REVIEW

**Review by ioanmo226 — COMMENTED** [review]
Screenshot_8
Seems like spam icon is distorted

**Comment by DenBond7** [comment]
intersting...

I have the following on my emulator

image

Could you do a comparison like I did?

**Comment by ioanmo226** [comment]
image

Maybe I thought too much but it seems distorted to me.

**Comment by ioanmo226** [comment]
Distored doesn't seem correct word. Spam icon is not seems to be centered in circle.

**Comment by DenBond7** [comment]
A few minutes. I will show you something new.

**Comment by DenBond7** [comment]
> Distored doesn't seem correct word. Spam icon is not seems to be centered in circle.

You were right. I've checked and centered the inner icon. Also, I've changed the whole image a little.

By the way, to check vertical and horizontal positions(and many other things) you can enable the Developer options. After that enable `Show layout bounds`
 
[Screen_recording_20230920_181057.webm](https://github.com/FlowCrypt/flowcrypt-android/assets/2863246/e6bc436b-1761-42b0-bf1c-d8e9a3a45ac6)

File: FlowCrypt/src/main/java/com/flowcrypt/email/jetpack/workmanager/base/BaseMoveMessagesWorker.kt [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+  private suspend fun moveMessagesInternal(
+    account: AccountEntity,
+    action: suspend (folderName: String, uidList: List<Long>) -> Unit
+  ) = withContext(Dispatchers.IO) {
+    val foldersManager = FoldersManager.fromDatabaseSuspend(applicationContext, account)
+    val folderAll = foldersManager.folderAll
+    val roomDatabase = FlowCryptRoomDatabase.getDatabase(applicationContext)
+
+    while (true) {
+      val candidatesForMoving = roomDatabase.msgDao().getMsgsWithStateSuspend(
```
Comment (ioanmo226) [inline FlowCrypt/src/main/java/com/flowcrypt/email/jetpack/workmanager/base/BaseMoveMessagesWorker.kt:87]: `messagesToMove` would be better naming

File: FlowCrypt/src/main/java/com/flowcrypt/email/jetpack/workmanager/base/BaseMoveMessagesWorker.kt [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+
+/**
+ * @author Denys Bondarenko
+ */
+abstract class BaseMoveMessagesWorker(context: Context, params: WorkerParameters) :
+  BaseSyncWorker(context, params) {
+
+  abstract val queryMessageState: MessageState
+  abstract suspend fun getDestinationFolderForIMAP(account: AccountEntity): LocalFolder?
+  abstract fun getAddAndRemoveLabelIdsForGmailAPI(srcFolder: String): Pair<List<String>?, List<String>?>
```
[the remark this answers is not in the document]
Comment (ioanmo226) [reply FlowCrypt/src/main/java/com/flowcrypt/email/jetpack/workmanager/base/BaseMoveMessagesWorker.kt]: Refactored code looks good :+1

**Review by ioanmo226 — APPROVED** [review]

**Pull request merged**